src/pk/ecc/ecc_set_dp_internal.c

```diff
@@ -0,0 +1,132 @@
+/* LibTomCrypt, modular cryptographic library -- Tom St Denis
+ *
+ * LibTomCrypt is a library that provides various cryptographic
+ * algorithms in a highly modular and flexible manner.
+ *
+ * The library is free for all purposes without any express
+ * guarantee it works.
+ */
+
+#include "tomcrypt.h"
+
+#ifdef LTC_MECC
+
+static int _ecc_cmp_hex_bn(const char *left_hex, void *right_bn)
+{
+   void *bn;
+   int match = 0;
+   if (mp_init(&bn) != CRYPT_OK)                    return 0;
```

Comment:
can you please add this as parameter so we don't have to do init&clear for each compare operation?

Reply:
done